sstables/mc/parquet_writer.hh

```diff
@@ -0,0 +1,1087 @@
+    {2, 1, UINT8{}}, // ROW_REGULAR_X_FLAGS,
+    {3, 1, INT64{}}, // ROW_REGULAR_X_TS,
+    {3, 1, INT32{}}, // ROW_REGULAR_X_LDT,
+    {3, 1, INT32{}}, // ROW_REGULAR_X_TTL,
+    {3, 1, UNKNOWN{}}, // ROW_REGULAR_X_VALUE,
+};
+
+struct cell_mapping {
+    int value;
+    parquet4seastar::logical_type::logical_type pq_type;
+    // Fields below are unused for keys
+    int flags;
+    int ts;
+    int ldt;
+    int ttl;
+};
+
+struct parquet_writer_schema {
+    int metadata_mappings[parts::ENUM_SIZE];
+    std::vector<cell_mapping> cell_mappings; // Indexed by ordinal id
+    parquet4seastar::writer_schema::schema p4s_schema;
+    const scylla_schema* scylla_sch;
+    int leaves;
+};
+
+bool is_supported_type(const parquet_writer_schema& pws, int ordinal_id) {
+    const auto& pq_type = pws.cell_mappings[ordinal_id].pq_type;
+    const auto& col_def = pws.scylla_sch->column_at((ordinal_column_id)ordinal_id);
+    return col_def.is_atomic()
+        && !std::holds_alternative<UNKNOWN>(pq_type);
+}
+
+
+parquet_writer_schema
+scylla_schema_to_parquet_writer_schema(const scylla_schema& scylla_sch) {
+    using namespace parquet4seastar;
+    using namespace parquet4seastar::writer_schema;
+    using namespace parquet4seastar::logical_type;
+    using namespace parts;
+
+    auto make_leaf = [] (std::string name, bool optional, logical_type::logical_type lt) {
+        primitive_node leaf{};
+        leaf.name = std::move(name);
+        leaf.optional = optional;
+        leaf.logical_type = lt;
+        leaf.compression = parquet4seastar::format::CompressionCodec::GZIP;
+        leaf.encoding = parquet4seastar::format::Encoding::RLE_DICTIONARY;
+        return leaf;
+    };
+
+    auto make_struct = [] (std::string name, bool optional) {
+        struct_node group{};
+        group.name = std::move(name);
+        group.optional = optional;
+        return group;
+    };
+
+    auto make_list = [] (std::string name, bool optional) {
+        list_node list{};
+        list.name = name;
+        list.optional = optional;
+        return list;
+    };
+
+    parquet_writer_schema pws;
+    pws.cell_mappings.resize(scylla_sch.all_columns().size());
+    pws.scylla_sch = &scylla_sch;
+
+    int leaf_idx = 0;
+
+    // Types
+    for (const auto& col_def : scylla_sch.all_columns()) {
+        int id = (int)col_def.ordinal_id;
+        logical_type::logical_type pq_type = map_type(col_def.type->get_kind());
+        pws.cell_mappings[id].pq_type = pq_type;
+    }
+    // header
+    {
+        auto header = make_struct("header", false);
+        // header_partition_key
+        {
+            auto header_partition_key = make_struct("partition_key", false);
+            for (const auto& col_def : scylla_sch.partition_key_columns()) {
+                int id = (int)col_def.ordinal_id;
+                logical_type::logical_type pq_type = pws.cell_mappings[id].pq_type;
+                if (!is_supported_type(pws, id)) {
```

Comment:
Silently dropping unsupported fields is no good - we need at least a logger warning to avoid problems in the future:
```cpp
parquet_logger.warn("Skipping unsupported type {} for column {}", col_def.type->name(), col_def.name_as_text());
```